pkg/client/client.go

```diff
@@ -1075,3 +1096,40 @@ func DoesLocalBucketExist(serverURL string, bucket string) (bool, error) {
 
 	return false, nil
 }
+
+func WaitMapVersionSync(timeout time.Time, smap dfc.Smap, idsToIgnore []string) error {
```

Comment:
If I get it right, the function does not wait anything. It checks the list of urls only once and sleeps between every urls. I thougth it would check all urls without sleeping in between and there would be outer loop that stops when url list is empty with sleep after inner loop finishes.

Reply:
There's a `continue` that will short-circuit each sleep on a successful check. The loop breaks when the map is empty.

Reply:
continue goes to the next item. So, when the loop reached end of the list it stops. What restarts the loop if anything is left in the list?

Reply:
The loop iterates over the map `urls`. With each successful check, it removes the corresponding url from the map. Sorry, I think I'm a little confused by your comment. Is this what you are asking about?